ethcore/src/client/client.rs

```diff
@@ -1745,6 +1745,14 @@ impl BlockChainClient for Client {
 		}
 	}
 
+	fn process_fork(&self) {
+		if self.importer.block_queue.processing_fork(&self.chain.read().best_block_hash()) {
+			debug!(target: "client", "Waiting for fork processing");
+			let _import_lock = self.importer.import_lock.lock();
+			debug!(target: "client", "Fork processing completed");
```

Comment:
How can we be certain that `processing` is now empty? Wouldn't it be better to call that code in a loop?